pkg/workload/pgx_helpers.go

```diff
@@ -69,6 +69,13 @@ func (p pgxLogger) Log(
 		// are expected and noisy.
 		return
 	}
+	// data may contain error with "restart transaction" -- skip those as well.
+	if data != nil {
+		ev := data["err"]
+		if err, ok := ev.(error); ok && strings.Contains(err.Error(), "restart transaction") {
```

Comment:
if you'd like, the check could be done once after formatting the message we log

i.e. check the resulting string from `fmt.Sprintf("pgx logger [%s]: %s logParams=%v", level.String(), msg, data)`

Reply:
That makes sense, I suppose; but this way it's more explicit... I'll keep it as is for now; but maybe we'll have to do this trick in the future.